File: webshop_frontend/src/app/models/product.ts
import { StandardModel } from "./standard-model";

export interface ProductUpload extends StandardModel {
    name: string;
    description: string;
    category: string;
    price: string;
    currency: string;
    image: Blob;
}

export interface Product extends StandardModel {
    name: string;
    description: string;
    category: string;
    price: number;
    currency: string;
    imageLocation: string;
}

File: webshop/src/main/java/com/example/demo/model/Product.java
package com.example.demo.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@Getter
@Setter
@Entity
public class Product {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "user_id")
	@NotNull
	private User user;

	@NotBlank
	private String name;

	@NotBlank
	private String description;

	@NotBlank
	private String category;

	@NotNull
	private Double price;

	@NotBlank
	private String currency;

	@NotBlank
	private String imageLocation;

}
